src/backend/distributed/executor/multi_executor.c

```diff
@@ -631,3 +648,36 @@ AlterTableConstraintCheck(QueryDesc *queryDesc)
 
 	return true;
 }
+
+
+/*
+ * IsLocalAndDistJoin returns true if the given plan belongs to a join
+ * between a local table and a distributed table.
+ */
+static bool
+IsLocalAndDistJoin(PlannedStmt *plannedStmt)
+{
+	bool hasLocalTables = false;
+	bool hasDistTables = false;
+	ListCell *oidCell = NULL;
+
+	if (plannedStmt->commandType != CMD_SELECT)
+	{
+		return false;
+	}
+
+	foreach(oidCell, plannedStmt->relationOids)
+	{
+		Oid relationId = lfirst_oid(oidCell);
+		if (RelationIsAKnownShard(relationId))
```

Comment:
No need to check if this is a reference table or not here. We wouldn't get here for non-reference distributed tables.